modules/system/assets/js/snowboard/extras/DataConfig.js

```diff
@@ -0,0 +1,208 @@
+     * attributes will be available as configuration values. This can be a security concern, so
+     * tread carefully.
+     *
+     * Otherwise, available configurations will be determined by the keys available in an object
+     * returned by a `defaults()` method in the instance.
+     *
+     * @returns {string[]|boolean}
+     */
+    getAcceptedConfigs() {
+        if (
+            this.instance.acceptAllDataConfigs !== undefined
+            && this.instance.acceptAllDataConfigs === true
+        ) {
+            return true;
+        }
+
+        if (
+            this.instance.defaults !== undefined
+            && typeof this.instance.defaults === 'function'
+            && typeof this.instance.defaults() === 'object'
+        ) {
+            return Object.keys(this.instance.defaults());
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns the default values for the instance.
+     *
+     * This will be an empty object if the instance either does not have a `defaults()` method, or
+     * the method itself does not return an object.
+     *
+     * @returns {object}
+     */
+    getDefaults() {
+        if (
+            this.instance.defaults !== undefined
+            && typeof this.instance.defaults === 'function'
+            && typeof this.instance.defaults() === 'object'
+        ) {
+            return this.instance.defaults();
+        }
+
+        return {};
+    }
+
+    /**
+     * Processes the configuration.
+     *
+     * Loads up the defaults, then populates it with any configuration values provided by the data
+     * attributes, based on the rules of the accepted configurations.
+     *
+     * This configuration object is then cached and available through `config.get()` calls.
+     *
+     * @returns {object}
+     */
+    processConfig() {
+        const config = this.getDefaults();
+
+        if (this.acceptedConfigs === false) {
+            return config;
+        }
+
+        /* eslint-disable */
+        for (const key in this.element.dataset) {
+            if (key === 'sticky') {
+                continue;
+            }
+            if (this.acceptedConfigs === true || this.acceptedConfigs.includes(key)) {
+                config[key] = this.coerceValue(this.element.dataset[key]);
+            }
+        }
+        /* eslint-enable */
+
+        return config;
+    }
+
+    /**
+     * Coerces configuration values for JavaScript.
+     *
+     * Takes the string value returned from the data attribute and coerces it into a more suitable
+     * type for JavaScript processing.
+     *
+     * @param {*} value
+     * @returns {*}
+     */
+    coerceValue(value) {
+        const stringValue = String(value);
+
+        // Null value
+        if (stringValue === 'null') {
+            return null;
+        }
+
+        // Undefined value
+        if (stringValue === 'undefined') {
+            return undefined;
+        }
```

Comment:
Can we also support base64 encoded data?

Reply:
What would be the intended purpose of that?

Reply:
Easier to include longer form data in the data attributes without having to worry about if a given value is going to escape the attribute quotes. May be needed by the inspector for more complex field types like richeditor.

Reply:
Unless we add some form of prefix or indicator that it is a base64 string, there's no way to tell from the outset if a given string is a base64 string or if it is indeed a legitimate value (like, say, a hash).

Reply:
How about we go with supporting the data:uri syntax for base64 https://en.wikipedia.org/wiki/Data_URI_scheme (down the road we can support other forms of Data URIs I suppose)?

Reply:
@LukeTowers I've added support for base64 - it will look for a `base64:` prefix in the config value.